src/main/java/uk/gov/companieshouse/accounts/association/tasks/DataMigrationPostProcessingTask.java

```diff
@@ -0,0 +1,94 @@
+package uk.gov.companieshouse.accounts.association.tasks;
+
+import java.util.stream.Collectors;
+import org.springframework.data.mongodb.core.query.Update;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.accounts.association.models.AssociationDao;
+import uk.gov.companieshouse.accounts.association.service.AssociationsService;
+import uk.gov.companieshouse.accounts.association.service.UsersService;
+import uk.gov.companieshouse.accounts.association.utils.StaticPropertyUtil;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+@Component
+public class DataMigrationPostProcessingTask {
+
+    private final AssociationsService associationsService;
+
+    private final UsersService usersService;
+
+    private static final Logger LOG = LoggerFactory.getLogger( StaticPropertyUtil.APPLICATION_NAMESPACE );
+
+    private static final String DATA_MIGRATION_USER_ID_UPDATE_TASK = "data-migration-user-id-update-task";
+
+    // TODO: choose different value e.g. 15
+    private static final int ITEMS_PER_PAGE = 2;
+
+    public DataMigrationPostProcessingTask( final AssociationsService associationsService, final UsersService usersService ) {
+        this.associationsService = associationsService;
+        this.usersService = usersService;
+    }
+
+    // TODO: change cron = "0 0 0 * * ?", so that the cron job runs everyday at midnight
+    @Scheduled( cron = "0 2 17 * * ?" )
+    public void processMigratedAssociations(){
+        LOG.infoContext(DATA_MIGRATION_USER_ID_UPDATE_TASK, "Starting up data migration user_id task...", null );
+
+        final var totalAssociations = associationsService.fetchNumberOfUnprocessedMigratedAssociations();
+        final var totalPages = ( totalAssociations / ITEMS_PER_PAGE ) + ( totalAssociations % 2 == 0 ? 0 : 1 );
+        int currentPage = (int) totalPages - 1;
+
+        LOG.debugContext(DATA_MIGRATION_USER_ID_UPDATE_TASK, String.format( "Identified %s associations to be processed in %s batches in reverse order... starting with batch %s", totalAssociations, totalPages, currentPage ), null );
+
+        while ( currentPage > -1 ) {
+            final var page = associationsService.fetchUnprocessedMigratedAssociations( currentPage, ITEMS_PER_PAGE );
```

Comment:
var page = his already gets all the associations where email_id exists.

Reply:
I'm not sure why we should remove `final`? The variable never gets updated.

The call to `fetchUnprocessedMigratedAssociations` will only get `ITEMS_PER_PAGE` number of associations on a given iteration e.g. if we set it to 100, it will fetch 100 migrating associations. Then on the next iteration it will fetch the next 100 etc. This lets us control how many are loaded into memory at the same time.